vertx-web/src/main/java/io/vertx/ext/web/impl/RouteState.java

```diff
@@ -444,6 +478,19 @@ RouteState addContextHandler(Handler<RoutingContext> contextHandler) {
       this.exclusive,
       this.exactPath);
 
+    int len = newState.contextHandlers.size();
+
+    if (len > 0) {
+      final Priority weight = weight(contextHandler);
+      Priority iterWeith = weight(newState.contextHandlers.get(len - 1));
+      if (iterWeith.ordinal() > weight.ordinal()) {
+        String message = "Cannot add [" + weight.name() + "] handler to route with [" + iterWeith.name() + "] handler at index " + (len - 1);
+        // when assertions are enabled, throw AssertionError to signal that the implementation is not correct
+        assert false : message;
```

Comment:
Given that assertions are not always active, we probably should guard this with a flag and throw?